### PR TITLE
Future allocation

### PR DESCRIPTION
**Pull request by jchyzy**

### CODE REVIEW

**Comment by PiekarzKarolina** [comment]
I would suggest moving the `Allocate nodes` button beside `Deploy notebook`. Or maybe even creating something like this:
image

And also I thing it would be good to change `Deploy notebook` to `Allocate nodes & deploy notebook`

File: gui/functionality/idact_notebook.py [hunk 1]
```diff
@@ -68,10 +63,90 @@ def __init__(self, data_provider, parent=None):
...
+        self.ui.begin_label.setDisabled(True)
+        self.ui.begin_edit.setDisabled(True)
+        self.ui.future_allocation.toggled.connect(self.change_buttons_to_disabled_or_not)
+
+    def change_buttons_to_disabled_or_not(self):
+        if self.ui.future_allocation.isChecked():
+            self.ui.deploy_button.setDisabled(True)
+            self.ui.begin_label.setDisabled(False)
+            self.ui.begin_edit.setDisabled(False)
+
```
Comment (PiekarzKarolina) [inline gui/functionality/idact_notebook.py]: I think here is unnecessary enter

**Comment by jchyzy** [comment]
I added Running Notebooks widget.

If it would be ok, things to add are:
- documentation
- properties in idact (not using private fields)

**Comment by pswedrak** [comment]
Please fix flake

File: gui/functionality/manage_jobs.py [hunk 2]
```diff
@@ -103,6 +120,73 @@ def show_jobs(self):
...
+            port_info = fetch_port_info(job_id, config)
+            [node_name, port] = port_info.split(" ")[0].split(":")
+
+            nodes_in_cluster[0].make_allocated(host=node_name,
+                                               port=int(port),
+                                               cores=None,
+                                               memory=None,
+                                               allocated_until=job.end_time)
+
+            entry_point_script_path = "~/.idact/entry_points"
```
Comment (PiekarzKarolina) [inline gui/functionality/manage_jobs.py]: I'm curious why this works on Windows

File: gui/functionality/running_notebooks.py [hunk 3]
```diff
@@ -15,6 +21,9 @@
 
 
 class RunningNotebooks(QWidget):
+    """ Module of GUI that is responsible for showing nodes and local ports for Jupyter Notebook (if exists),
+    cancelling jobs, opening notebooks in the browser if they exists or deploy them if not.
```
Comment (PiekarzKarolina) [inline gui/functionality/running_notebooks.py]: typo deploys

File: gui/helpers/deployments_provider.py [hunk 4]
```diff
@@ -39,6 +49,10 @@ def add_deployment(self, cluster_name, deployment):
             self.get_deployments(cluster_name).dask_deployments.append(deployment)
 
     def _update_deployments(self, cluster_name: str):
+        """
+        Update deployments for the cluster.
```
Comment (PiekarzKarolina) [inline gui/helpers/deployments_provider.py]: typo updates

File: gui/functionality/running_notebooks.py (hunk 3 above)
Comment (jchyzy) [reply gui/functionality/running_notebooks.py]: Fixed to deploying

File: gui/helpers/deployments_provider.py (hunk 4 above)
Comment (jchyzy) [reply gui/helpers/deployments_provider.py]: Fixed

**Pull request merged**